### ISSUE TITLE
[bug]: Keploy is unable to handle high QPS

### ISSUE THREAD

**Issue by shivamsouravjha** [keploy/keploy#1314]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

* Incase of high QPS the data that is written via AddDataEvent, before picked via receiver is updated (as another data event is appended) leading to malformed request error.
* Handling of global race condition via using queues might help in dealing with such scenarios.

### Steps to reproduce

* With mutex concurrent read write is avoided but two goroutines updating same map leads to undesired change in data.

### Environment

None

### Version

Cloud

### Repository

keploy

**Comment by charankamarapu** [comment]
Please provide proper steps for reproducing the issue.

**Comment by Krishnan9074** [comment]
Hi @shivamsouravjha is this issue solved already?

**Comment by shivamsouravjha** [comment]
hey @Krishnan9074 this issue needs major code and architecture chagnes that team has already put in the project roadmap, thus created the issue.

**Comment by Krishnan9074** [comment]
Yeah that's what I felt, is there anything on there u feel i could contribute to?
